cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,534 @@
+#!/usr/bin/env python3
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Cylc configuration linter.
+
+Checks code style, deprecated syntax and other issues."""
+# NOTE: docstring needed for `cylc help all` output
+# (if editing check this still comes out as expected)
+
+COP_DOC = """cylc lint [OPTIONS] ARGS
+
+Cylc lint looks through one or more folders for ".cylc" and ".rc"
+files.
+
+Checks code style, deprecated syntax and other issues.
+
+Can be run either as a linter or "in place" ("-i"), leaving comments
+in files.
+
+When run with "-i" ("--inplace") mode this will alter your files.
+We strongly recommend committing your workflow to version control
+before using "cylc lint -i".
```

Comment:
```suggestion
In-place mode ( "-i/ --inplace") writes suggestions into the file as comments. Commit to your
version control system before using this, in case you want to back out the changes quickly.
```